staff/oriol-bover/array-functions/splice.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Abstraction of splyce
+ * 
+ * Iterate an array and  changes the contents of an array by removing or replacing existing elements and/or adding new elements.
+ * 
+ * @param {Array} array 
+ * @param {Number} start 
+ * @param {Number} deleteCount 
+ * 
+ * @returns {Array} - An array containing the deleted elements.
+ */
+
+
+function splice(array, start, deleteCount) {
+    if (!(array instanceof Array)) throw TypeError(array + ' should be an Array');
+    if (typeof start !== 'number') throw TypeError(start + ' should be a Number');
+    if (typeof deleteCount !== 'number') throw TypeError(deleteCount + ' should be a Number');
+
+    start = start > array.length ? array.length : (start >= 0 ? start : Math.max(0, array.length + start));
+    deleteCount = typeof deleteCount === 'undefined' || deleteCount > array.length ? array.length : (deleteCount > 0 ? deleteCount : 0);
+    
+    var array2 = Object.assign([], array);    
```

Comment:
I am afraid this is ECMAScript 2015 (6th Edition, ECMA-262)...
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

You could use a for loop to iterate and copy the original array